File: 238-product-of-array-except-self/product-of-array-except-self.cpp
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
        int n=nums.size();
        int left=1;
        int right=1;
        int i=0;
        int j=n-1;
        vector<int>ans(n,1);
        while(i<n)
        {
            ans[i]*=left;
            left*=nums[i];

            ans[j]*=right;
            right*=nums[j];

            i++;
            j--;
        }
        return ans;
    }
};

File: 238-product-of-array-except-self/product-of-array-except-self.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        left=1
        right=1
        i=0
        n=len(nums)
        j=n-1
        ans=[1]*n
        while i<n:
            ans[i]*=left
            left*=nums[i]

            ans[j]*=right
            right*=nums[j]

            i+=1
            j-=1
        return ans
            
        